extensions/podman/packages/extension/src/utils/podman-cli.spec.ts

```diff
@@ -63,12 +65,26 @@ describe('findPodmanInstallations', () => {
     vi.mocked(env).isLinux = false;
 
     // Setup default fs mocks
-    vi.mocked(fs.lstatSync).mockReturnValue({
-      isFile: vi.fn().mockReturnValue(true),
-      isSymbolicLink: vi.fn().mockReturnValue(false),
-      // Cast through 'unknown' because fs.Stats has many properties we don't need to mock
-      // We only mock the methods actually used in the implementation (isFile, isSymbolicLink)
-    } as unknown as fs.Stats);
+    vi.mocked(fs.realpathSync).mockImplementation(path => {
+      if (path === '/symlink/podman') {
+        return '/opt/podman/bin/podman'; // symlink points here
+      }
+      return path as string; // not a symlink, return as-is
+    });
+
+    // Mock statSync for inode checking
+    vi.mocked(fs.statSync).mockImplementation(path => {
+      let ino = randomInt(1000000, 9999999);
```

Comment:
question(blocking): we should not introduce randomness in testing logic, can't we just use a given value?